rfcs/active/0000-badge.md

```diff
@@ -0,0 +1,362 @@
+<oui-badge>New</oui-badge>
+
+<!-- Badge with variant -->
+<oui-badge variant="primary">4</oui-badge>
+
+<!-- Pill badge -->
+<oui-badge shape="pill">New</oui-badge>
+
+<!-- Badge in context with another element -->
+<h2>Product Features <oui-badge>New</oui-badge></h2>
+
+<!-- Badge as a counter -->
+<oui-button>
+  Notifications
+  <oui-badge variant="secondary">4</oui-badge>
+</oui-button>
+
+<!-- Positioned badge (for notification indicators) -->
+<oui-button>
+  Inbox
+  <oui-badge position="top-end" variant="danger">99+</oui-badge>
+</oui-button>
+
+<!-- Dot badge (for status indicators) -->
+<oui-badge dot aria-label="New notifications"></oui-badge>
+
+<!-- Status badge -->
+<oui-badge status="success">Success</oui-badge>
+
+<!-- Badge with custom color -->
+<oui-badge color="#52c41a">Custom</oui-badge>
+
+<!-- Badge with overflow count -->
+<oui-badge count="125" overflow-count="99"></oui-badge>
+
+<!-- Standalone ribbon badge -->
+<oui-badge.ribbon text="Ribbon" placement="end">
+  <div style="width: 300px; height: 100px; background: #f0f0f0;"></div>
+</oui-badge.ribbon>
+```
+
+# Motivation
+
+Badges are a fundamental UI pattern found in many digital interfaces across various platforms. They serve several key purposes:
+
+1. **Information density** - Badges allow developers to display additional information (counts, status, labels) in a compact space.
+2. **Visual indication** - They provide immediate visual cues to users about state, notifications, or important updates.
+3. **Context enhancement** - They enhance other components by providing supplementary information (counters, status indicators).
+
+The OpenUI Design System aims to provide a comprehensive set of common UI components, and badges represent a widely-used pattern that fits within this mission. Adding a standardized badge component would reduce duplicative effort in rebuilding this common UI pattern across different applications and frameworks.
+
+# Detailed Proposal
+
+The `oui-badge` component will provide a versatile, accessible, and themeable implementation of the badge pattern.
+
+## API
+
+### Properties
+
+| Property | Type | Default | Description |
+|----------|------|---------|-------------|
+| `variant` | `'primary' \| 'secondary' \| 'success' \| 'danger' \| 'warning' \| 'info'` | `'primary'` | Determines the visual style of the badge |
+| `shape` | `'default' \| 'pill'` | `'default'` | Controls the border-radius of the badge |
+| `position` | `null \| 'top-start' \| 'top-end' \| 'bottom-start' \| 'bottom-end'` | `null` | When used with a parent element, positions the badge accordingly |
+| `visible` | `boolean` | `true` | Controls whether the badge is displayed |
+| `aria-label` | `string` | `null` | Accessible label (required when badge only contains numbers or non-descriptive content) |
+| `color` | `string` | `null` | Custom color for the badge (hex, rgb, or named color) |
+| `count` | `number \| string` | `null` | Number or content to show in the badge |
```

Comment:
Why can this be specified both via a property and as child content? Seems like just as a child makes sense?